prefect_gcp/workers/vertex.py

```diff
@@ -0,0 +1,633 @@
+        to complete.
+
+        Args:
+            flow_run: The flow run to execute
+            configuration: The configuration to use when executing the flow run.
+            task_status: The task status object for the current flow run. If provided,
+                the task will be marked as started.
+
+        Returns:
+            VertexAIWorkerResult: A result object containing information about the
+                final state of the flow run
+        """
+        logger = self.get_flow_run_logger(flow_run)
+
+        client_options = ClientOptions(
+            api_endpoint=f"{configuration.region}-aiplatform.googleapis.com"
+        )
+
+        job_name = configuration.job_name
+
+        job_spec = self._build_job_spec(configuration)
+        with configuration.credentials.get_job_service_client(
+            client_options=client_options
+        ) as job_service_client:
+            job_run = await self._create_and_begin_job(
+                job_name, job_spec, job_service_client, configuration, logger
+            )
+
+            if task_status:
+                task_status.started(job_name)
+
+            final_job_run = await self._watch_job_run(
+                job_name=job_name,
+                full_job_name=job_run.name,
+                job_service_client=job_service_client,
+                current_state=job_run.state,
+                until_states=(
+                    JobState.JOB_STATE_SUCCEEDED,
+                    JobState.JOB_STATE_FAILED,
+                    JobState.JOB_STATE_CANCELLED,
+                    JobState.JOB_STATE_EXPIRED,
+                ),
+                configuration=configuration,
+                logger=logger,
+                timeout=int(
+                    datetime.timedelta(
+                        hours=configuration.job_spec["maximum_run_time_hours"]
+                    ).total_seconds()
+                ),
+            )
+
+        error_msg = final_job_run.error.message
+        if error_msg:
+            raise RuntimeError(error_msg)
+
+        status_code = 0 if final_job_run.state == JobState.JOB_STATE_SUCCEEDED else 1
+
+        return VertexAIWorkerResult(
+            identifier=final_job_run.display_name, status_code=status_code
+        )
+
+    def _build_job_spec(
+        self, configuration: VertexAIWorkerJobConfiguration
+    ) -> "CustomJobSpec":
+        """
+        Builds a job spec by gathering details.
+        """
+        # gather worker pool spec
+        worker_pool_specs = []
+        for spec in configuration.job_spec["worker_pool_specs"]:
+            worker_pool_specs.append(
+                WorkerPoolSpec(
+                    container_spec=ContainerSpec(**spec["container_spec"]),
+                    machine_spec=MachineSpec(**spec["machine_spec"]),
+                    replica_count=spec["replica_count"],
+                    disk_spec=DiskSpec(**spec["disk_spec"]),
+                )
+            )
```

Comment:
Could potentially be written as a comprehension:
```suggestion
        worker_pool_specs = [
            WorkerPoolSpec(                 
                 container_spec=ContainerSpec(**spec["container_spec"]),
                 machine_spec=MachineSpec(**spec["machine_spec"]),
                 replica_count=spec["replica_count"],
                 disk_spec=DiskSpec(**spec["disk_spec"]),
              ) for spec in configuration.job_spec["worker_pool_specs"]
        ]
```